src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -225,7 +210,8 @@ async Task EnsureLong()
                 }
                 finally
                 {
-                    combinedCts?.Dispose();
+                    if (async && Timeout > TimeSpan.Zero)
```

Comment:
It's not amazing that we check for this twice. We could have an internal check inside _timeoutCts, which would mean it should be legal to call Stop twice (i.e. remove the Assert).

Reply:
I completely agree that it is a bit sad that we need to check twice.
When you asked for the consistency checks I was already aware of this problem but intentionally opted for the maximum variant of consistency checking.
We can certainly remove the assertions from `Stop()` and by that declare it as legal to call it twice or in situations where there was no corresponding call to `Start()`.
I don't understand what you have in mind regarding an "internal check inside _timeoutCts".